WordPress/Classes/ViewRelated/Site Creation/Final Assembly/SiteAssemblyContentView.swift

```diff
@@ -141,8 +135,12 @@ final class SiteAssemblyContentView: UIView {
 
     let siteCreator: SiteCreator
 
+    var isFreeDomain: Bool?
     private lazy var shouldShowDomainPurchase: Bool = {
-        siteCreator.shouldShowDomainCheckout
+        if let isFreeDomain = isFreeDomain {
+            return isFreeDomain
+        }
+        return siteCreator.shouldShowDomainCheckout
```

Comment:
I want to point out that this logic shows the "Paid" copy for free domains. I think you meant this:

```swift
if let isFreeDomain = isFreeDomain {
    return !isFreeDomain
}
```

Also, if `isFreeDomain` is not optional and `false`, the `shouldShowDomainPurchase` will be `true` without taking into consideration the feature flag and AB test conditions in `siteCreator.shouldShowDomainCheckout`.  To fix that, I think we can do the following:

```swift
if let isFreeDomain = isFreeDomain, isFreeDomain {
    return !isFreeDomain
}
```

This code means to return `return !isFreeDomain` only if the domain is free. But if it's paid, the `siteCreator.shouldShowDomainCheckout` should be executed instead.